neonvm/controllers/virtualmachine_controller.go

```diff
@@ -609,24 +609,24 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 		}
 
 		// runner pod found, check that it's still up:
-		switch vmRunner.Status.Phase {
-		case corev1.PodSucceeded:
+		switch runnerContainerStatus(vmRunner) {
```

Comment:
This function is in dire need of refactoring. 500+ lines and nested switch cases make it hard to follow, and e.g. to understand which `case`/`default` applies to which `switch`.

Reply:
see #733